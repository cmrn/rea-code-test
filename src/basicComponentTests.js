import React from 'react';
import { shallow } from 'enzyme';

export default function basicComponentTests(component) {
  it('allows a className to be set', () => {
    const wrapper = shallow(React.cloneElement(component, { className: "classy" }));
    expect(wrapper.props().className).toEqual(expect.stringContaining('classy'));
  });

  it('passes through other props', () => {
    const wrapper = shallow(React.cloneElement(component, { foo: "bar" }));
    expect(wrapper.props().foo).toEqual('bar');
  });
}
