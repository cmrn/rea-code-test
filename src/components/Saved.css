.Saved-hover {
  color: white;
  font-size: 25px;
  text-align: center;
}

.Saved-hover-icon {
  width: 75px;
  height: 75px;
}