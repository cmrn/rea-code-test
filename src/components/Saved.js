import React from 'react';
import RemoveIcon from 'react-icons/lib/md/clear';
import PropertyList from '../components/PropertyList';
import './Saved.css';

const Saved = (props) => (
  <PropertyList 
    {...props}
    title='Saved Properties' 
    hoverContent={
      <div className='Results-hover'>
        <RemoveIcon className='Results-hover-icon' />
        <div>Remove</div>
      </div>
    }
  />
);

export default Saved;
