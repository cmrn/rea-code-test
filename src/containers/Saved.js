import React from 'react';
import { view } from 'react-easy-state';
import Saved from '../components/Saved';
import propertyStore from '../propertyStore';

const SavedContainer = (props) => (
  <Saved 
    {...props} 
    propertyIds={propertyStore.saved}
    onClick={(id) => propertyStore.unsave(id)}
  />
);

export default view(SavedContainer);
