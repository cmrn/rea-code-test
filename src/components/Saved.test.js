import React from 'react';
import basicComponentTests from '../basicComponentTests';
import { shallow } from 'enzyme';
import Saved from './Saved';

const dummyProps = {
  propertyIds: []
};

basicComponentTests(<Saved {...dummyProps} />);

it('sets title and hover content as expected', () => {
  const wrapper = shallow(<Saved {...dummyProps} />);
  expect(wrapper).toMatchSnapshot();
});