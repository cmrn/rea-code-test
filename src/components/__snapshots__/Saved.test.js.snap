// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets title and hover content as expected 1`] = `
<PropertyList
  hoverContent={
    <div
      className="Results-hover"
    >
      <MdClear
        className="Results-hover-icon"
      />
      <div>
        Remove
      </div>
    </div>
  }
  propertyIds={Array []}
  title="Saved Properties"
/>
`;
